ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/spanning-tree.j2

```diff
@@ -9,6 +9,43 @@ spanning-tree mode {{ spanning_tree.mode }}
 no spanning-tree vlan-id {{ spanning_tree.no_spanning_tree_vlan }}
 {%     endif %}
 {%     if spanning_tree.mode == "mstp" %}
-spanning-tree mst 0 priority {{ spanning_tree.priority }}
+{%         if spanning_tree.mst_instances is defined and spanning_tree.mst_instances is not none  %}
+{%             for mst_instance in spanning_tree.mst_instances | arista.avd.natural_sort %}
+spanning-tree mst {{ mst_instance }} priority {% if spanning_tree.mst_instances[mst_instance].priority is defined and spanning_tree.mst_instances[mst_instance].priority is not none %}{{ spanning_tree.mst_instances[mst_instance].priority }}
+{%                 endif %}
+{%             endfor %}
+{%         endif %}
+{%     elif spanning_tree.mode == "rapid-pvst" %}
+{%         if spanning_tree.rapid_pvst_instances is defined and spanning_tree.rapid_pvst_instances is not none  %}
+{%             for vlan_id in spanning_tree.rapid_pvst_instances | arista.avd.natural_sort %}
+spanning-tree vlan-id {{ vlan_id }} priority {% if spanning_tree.rapid_pvst_instances[vlan_id].priority is defined and spanning_tree.rapid_pvst_instances[vlan_id].priority is not none %}{{ spanning_tree.rapid_pvst_instances[vlan_id].priority }}
+{%                 endif %}
+{%             endfor %}
+{%         endif %}
+{%     else %}
+{%         if spanning_tree.rstp_priority is defined and spanning_tree.rstp_priority is not none  %}
+spanning-tree priority {{ spanning_tree.rstp_priority }}
+{%         endif %}
 {%     endif %}
-{% endif %}
\ No newline at end of file
+{%     if spanning_tree.mst is defined and spanning_tree.mst is not none %}
+{%         if spanning_tree.mst.pvst_border is defined and spanning_tree.mst.pvst_border == true %}
+spanning-tree mst pvst border
+{%         endif %}
+{%         if spanning_tree.mst.configuration is defined and spanning_tree.configuration is not none %}
+!
+spanning-tree mst configuration
+{%             if spanning_tree.mst.configuration.name is defined and spanning_tree.configuration.name is not none %}
+   name {{ spanning_tree.mst.configuration.name }}
+{%             endif %}
+{%             if spanning_tree.mst.configuration.revision is defined and spanning_tree.configuration.revision is not none %}
+   revision {{ spanning_tree.mst.configuration.revision }}
+{%             endif %}
+{%             if spanning_tree.mst.configuration.instances is defined and spanning_tree.configuration.instances is not none %}
+{%                 for instance in spanning_tree.mst.configuration.instances | arista.avd.natural_sort %}
+   instance {{ instance }} vlan {% if spanning_tree.mst.configuration.instances[instance].vlans is defined and spanning_tree.mst.configuration.instances[instance].vlans is not none %}{{ spanning_tree.mst.configuration.instances[instance].vlans }}
```

Comment:
You will need to put the `if` before the config line. Otherwise you may end up rendering `instance 1 vlan` if the priority keyword is missing.

Reply:
Very good point, fixed it.

Reply:
Good point, fixed!